docs/api.rst

```diff
@@ -1,93 +1,87 @@
-Interfaces, exceptions and utils
-================================
+API
+===
 
-Connection Clients
-------------------
+The naming of things can be confusing when using Bluetooth LE, due to the many possible meanings of
+words like "device" or "service" depending on the context. Here is a quick breakdown of the main object
+types used in bleak and their meaning, roughly in the order in which you will often use them in your code:
```

Comment:
This is a really nice way to introduce the API.

Reply:
I've added a few notes on attributes and handles too.

Reply:
The purpose of handles in Bleak is to differentiate between two attributes with the same UUID. More info in #888.

Reply:
Ok, rephrased.